tests/instancetype/instancetype.go

```diff
@@ -596,18 +595,14 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			newVM := libvmi.NewVirtualMachine(newVMI,
 				libvmi.WithInstancetype(instancetype.Name),
 				libvmi.WithPreference(preference.Name),
+				libvmi.WithRunStrategy(virtv1.RunStrategyAlways),
 			)
 			newVM, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), newVM, metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
-
-			By("Waiting for a VirtualMachineInstancetypeSpec ControllerRevision to be referenced from the new VirtualMachine")
-			Eventually(func() string {
-				newVM, err = virtClient.VirtualMachine(newVM.Namespace).Get(context.Background(), newVM.Name, metav1.GetOptions{})
-				if err != nil {
-					return ""
-				}
-				return newVM.Spec.Instancetype.RevisionName
-			}, 300*time.Second, 1*time.Second).ShouldNot(BeEmpty())
+			By("Waiting for VM to be ready")
+			EventuallyWithOffset(1, matcher.ThisVM(newVM), 360*time.Second, 1*time.Second).Should(matcher.BeReady())
+			newVM, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(newVM)).Get(context.Background(), newVM.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
With this change, we lost the check that the `newVM.Spec.Instancetype.RevisionName` is not empty  and the `Expect(newVM.Spec.Instancetype.RevisionName).ToNot(Equal(vm.Spec.Instancetype.RevisionName))` is not strong enough.
I think we should explicitly check test the non emptiness

Reply:
Yes, but there is no guarantee that `newVM.Spec.Instancetype.RevisionName` is not empty and hence `Expect(newVM.Spec.Instancetype.RevisionName).ToNot(Equal(vm.Spec.Instancetype.RevisionName))` passes

Reply:
I think that if `newVM.Spec.Instancetype.Name` is not empty, then
`Expect(newVM.Spec.Instancetype.RevisionName).ToNot(Equal(vm.Spec.Instancetype.RevisionName))` will fail because we know for sure that `vm.Spec.Instancetype.RevisionName` is not empty due to the check above:
`Expect(vm.Spec.Instancetype.RevisionName).To(ContainSubstring(instancetype.Name))`.

However, I don't see any harm in adding this additional check anyway, so I'll include it as well.

Reply:
Just for completeness, because maybe I am not explaining myself in the proper way:
The issue is that we are not checking that `newVM.Spec.Instancetype.Name` is not empty.
IOW, if it is empty there could be a problem because in this case 
 ```
 Expect(newVM.Spec.Instancetype.RevisionName).ToNot(Equal(vm.Spec.Instancetype.RevisionName))
 ```
 would result in
 ```
  Expect(").ToNot(Equal(vm.Spec.Instancetype.RevisionName))
 ```
 which is true, but wrong :)

Reply:
But it's `To` and not `ToNot`

Reply:
yes you are right i was looking into:
https://github.com/Barakmor1/kubevirt/blob/5415ab026e9db15e43d4954430789d79db8c6492/tests/instancetype/instancetype.go#L609
Sorry for the noise :)

Reply:
All good! I want to be sure I am not missing anything. Thank you! :)